azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/ConfigurationParser.java

```diff
@@ -54,18 +65,25 @@ protected String getResourceGroup() throws MojoExecutionException {
 
     protected abstract WebContainer getWebContainer() throws MojoExecutionException;
 
-    protected abstract List<Resource> getResources();
+    protected abstract List<Resource> getResources() throws MojoExecutionException;
 
     public WebAppConfiguration getWebAppConfiguration() throws MojoExecutionException {
         WebAppConfiguration.Builder builder = new WebAppConfiguration.Builder();
-        if (OperatingSystemEnum.Windows == getOs()) {
-            builder = builder.javaVersion(getJavaVersion()).webContainer(getWebContainer());
-        }
-        if (OperatingSystemEnum.Linux == getOs()) {
-            builder = builder.runtimeStack(getRuntimeStack());
-        }
-        if (OperatingSystemEnum.Docker == getOs()) {
-            builder = builder.image(getImage()).serverId(getServerId()).registryUrl(getRegistryUrl());
+        final OperatingSystemEnum os = getOs();
+        if (os != null) {
```

Comment:
What if `os == null`

Reply:
`os == null` means users do not specify any runtime related configurations, which is allowed.

Reply:
Will add a debug log when `os==null`.